blockchain/blockchain.go

```diff
@@ -2128,8 +2141,17 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 		return nil, 0, err
 	}
 	if vmConfig.EnableInternalTxTracing {
-		// TODO-InternalTxTracer
-		// Need to store traces to the database or send them to the designated channel
+		switch tracer := vmConfig.Tracer.(type) {
+		case *vm.InternalTxTracer:
+			result, err := tracer.GetResult()
+			if err != nil {
+				logger.Error("failed to get tracing result from a transaction", "txHash", tx.Hash().String(), "err", err)
+				return nil, 0, err
```

Comment:
Did you consider that this might block block-processing, even if the block itself does not have any issue?

Reply:
Yes. I think turning on `--vm.internaltx` mean that the user wants to gather the tracing results. So, it has to return an error although it shouldn't happen. What do you think?